app/deployments/routes.py

```diff
@@ -1451,11 +1542,155 @@ def createdep():
+    str_retry = " retry_"
+    tmp_name = dep.description.split(str_retry)[0]
+    deps = dbhelpers.get_user_deployments(session["userid"])
+    
+    for tmp_dep in deps:
+        if (tmp_name + str_retry in tmp_dep.description and 'DELETE' not in tmp_dep.status):
+            num_retry = int(tmp_dep.description.split(str_retry)[1])
+            
+            if num_retry > max_num_retry:
+                max_num_retry = num_retry
+                
+    additionaldescription = tmp_name + str_retry + str(max_num_retry + 1)
+
+    source_template = tosca_info[dep.selected_template]
+    form_data = inputs
+    template, template_text = load_template(dep.selected_template)
+
+    uuidgen_deployment = str(uuid_generator.uuid1())
+
+    doprocess, inputs, stinputs = process_inputs(
+        source_template, inputs, form_data, uuidgen_deployment
+    )
+
+    # If input is a bucket_name check for validity
+    for name in inputs:
+        if search("bucket_name", name):
+            errors = check_s3_bucket_name(uuidgen_deployment + "-" + inputs[name])
+
+            if errors:
+                for error in errors:
+                    flash(error, "danger")
+                return redirect(url_for(SHOW_DEPLOYMENTS_ROUTE))
+
+    app.logger.debug(f"Calling orchestrator with inputs: {inputs}")
+
+    if doprocess:
+        storage_encryption, vault_secret_uuid, vault_secret_key = (
+            add_storage_encryption(access_token, inputs)
+        )
+        params = {}  # is it needed??
+        create_deployment(
+            template,
+            inputs,
+            stinputs,
+            form_data,
+            dep.selected_template,
+            source_template,
+            template_text,
+            additionaldescription,
+            params,
+            storage_encryption,
+            vault_secret_uuid,
+            vault_secret_key,
+        )
+
+        flash(
+            f"Retry action for deployment {dep.description} <{depid}> successfully triggered!",
+            "success",
+        )
+
+    return redirect(url_for(SHOW_DEPLOYMENTS_ROUTE))
+
+
+def check_s3_bucket_name(name):
+    """
+    Validates an S3 bucket name based on the given rules.
+
+    :param name: The bucket name to validate.
+    :return: A tuple (bool, list) where the bool indicates if the name is valid,
+             and the list contains validation error messages (if any).
+    """
+    errors = []
+
+    # Rule 1: Length between 3 and 63 characters
+    if not (3 <= len(name) <= 63):
+        errors.append("Bucket names must be between 3 and 63 characters long.")
+
+    # Rule 2: Allowed characters
+    if not match(r"^[a-z0-9.-]+$", name):
+        errors.append(
+            "Bucket names can only contain lowercase letters, numbers, dots (.), and hyphens (-)."
+        )
+
+    # Rule 3: Begin and end with a letter or number
+    if not match(r"^[a-z0-9].*[a-z0-9]$", name):
+        errors.append("Bucket names must begin and end with a letter or number.")
```

Comment:
Qua in realtà l'inizio del nome del bucket è appunto un uuid quindi il nome scelto dall'utente può non per forza iniziare con una lettera o numero